src/main/java/hudson/plugins/logparser/LogParserPublisher.java

```diff
@@ -203,15 +213,15 @@ public boolean getLegacyFormatting() {
         }
         
         public String getColorError() {
-            return colorError;
+            return colorError == null || colorError.isEmpty() ? defaultErrorColor : colorError;
```

Comment:
You could just check it once and then set the colorError/colorWarning/colorInfo to the defaultErrorColor, etc.